### ISSUE TITLE
Different desktop on `xrandr HDMI --left-of eDP`

### ISSUE THREAD

**Issue by Lang-Lovdog** [yazeed1s/zwm#29]
## Issue Type
- [ *] Bug Report
- [ ] Feature Request
- [ ] Enhancement

## Description
zwm places every opened program (except for polybar) only on HDMI when using HDMI side of laptop's screen

## Steps to Reproduce (for Bug Reports)
1.  Open zwm
2.  Connect HDMI
3.  Configure with `xrandr --output HDMI --left-of eDP1`
4. Open windows and change desktops

## Expected Behavior
Each screen corresponding to a different desktop and can move windows between them.

## Actual Behavior
zwm places all opened windows in one screen only and ignore the second leaving only polybar

## Logs and Screenshots
Attach any relevant logs or screenshots that might help in diagnosing the issue.

## Additional Information
This happens even if I start zwm before connecting the external monitor or after.
If I connect the external before, X config defaults to `HDMI --same-as eDP`

## Checklist
- [* ] I have searched for duplicate issues to ensure this is not a duplicate.
---

**Comment by yazeed1s** [comment]
If you’re connecting your monitor before starting ZWM and still running into this issue, then it might be a bug or something with xrandr. But, currently, zwm doesn’t pick up on monitors being connected or disconnected while it’s running, it just picks up monitors that were connected before it started (I didn't handle cases where monitors change on the fly). I’ll fix this behavior next week or the week after when I get the time for it.

**Comment by yazeed1s** [comment]
@Lang-Lovdog I have added `xrandr HDMI --left-of eDP` to my `.xinirc` and the issue cannot be reproduced. It works as expected but the status bar shows up in the primary monitor only. Note, i did all of this before starting zwm.

**Comment by yazeed1s** [comment]
Ignore my previous reply, your issue seems to happen only if you connect the external monitor after starting zwm. I will fix that

**Comment by yazeed1s** [comment]
@Lang-Lovdog the bug should now be fixed in #32. I'll keep the PR open for a few days to test the changes thoroughly before merging into the master branch. In the meantime, feel free to pull the changes from #32 and report any bugs you encounter.

**Comment by Lang-Lovdog** [comment]
Thanks, sorry for not answering before. I was kinda busy with some school things.

I'll test it ASAP.

**Comment by Lang-Lovdog** [comment]
> @Lang-Lovdog the bug should now be fixed in #32. I'll keep the PR open for a few days to test the changes thoroughly before merging into the master branch. In the meantime, feel free to pull the changes from #32 and report any bugs you encounter.
---
I'm trying it rn, the feature works pretty well and haven't found any bug at the moment.
---
Changing the topic a little bit, in order to switch monitor focus without mouse, is it possible to map it?

**Comment by yazeed1s** [comment]
Okay that's cool, I have found few bugs and most of them were fixed. i am still fixing some other deep issues related to the tree. The feature you requested seems easy to implement, i would suggest you open another issue and request it so i can keep things intact in this repo.

**Comment by Lang-Lovdog** [comment]
> Okay that's cool, I have found few bugs and most of them were fixed. i am still fixing some other deep issues related to the tree. The feature you asked for isn't implemented into the wm, but it seems easy to implement, i would suggest you open another issue and request it so i can keep things organized in this repo. 

Thanks, I'll open the request.

About some bugs I found (20min. ago) are actually just about the tree as you mentioned above, but I don't know how to describe them, I'm really sorry 🙇🏾 if my description is not clear.

* There's more than one window open
* I close/move one of the Windows
* The Windows left got buggy within desktops with the same desktop number (get kinda misplaced in both screens).
* If not a buggy window or when it fixes itself, the second screen stops receiving Keyboard events and zwm instructions.

But, the last one happens too if my monitor turns off and on without xrandr changing the layouts.

Hope this helps, maybe you've noticed this yet, but better to be sure.
Thanks again for your time and work 😸✨

**Comment by yazeed1s** [comment]
Interesting! I believe it's connected to a deep-rooted bug I’ve been working on in the tree. It occurs when windows are moved or deleted, and it results in tree corruption caused by dangling pointers. This impacts the desktop trying to hold the newly moved node. After tracing the issue, I pinpointed it to how windows are unlinked from the tree. I’ve rewritten the unlink_node function and just pushed the changes to #32. Please check if the bug you encountered still persists. I’ll also do thorough testing on my end. Thanks for reporting this!

**Comment by yazeed1s** [comment]
Looks like the latest changes in #32 added more bugs. I am tracking some double frees and other memory curruptions that happen after disconnected monitors are merged.

**Comment by yazeed1s** [comment]
@Lang-Lovdog I have fixed all the bugs i encountered. This should resolve your issue with the buggy desktops and windows. Lemme know if things are good from your side and your issue was fixed so i can merge to main.

**Comment by Lang-Lovdog** [comment]
@yazeed1s It seems that I lost click event inside windows. I can only select window but can't interact.

**Comment by yazeed1s** [comment]
Is focous_follow_pointer set to false?

**Comment by yazeed1s** [comment]
my bad, i was expermenting with some changes in that function, and forgot to restore the old code. I will fix that rn and push it to #32

**Comment by Lang-Lovdog** [comment]
> Is focous_follow_pointer set to false?

Yep, it is set to false. Just because of some floating window things I've been experimenting with.

**Comment by Lang-Lovdog** [comment]
> my bad, i was expermenting with some changes in that function, and forgot to restore the old code. I will fix that rn and push it to #32 

I have no problem if you still need to get it altered. I'll just put my config to true.
